### ISSUE TITLE
Recall trends to 0

### ISSUE THREAD

**Issue by ZacharyForrest** [experiencor/keras-yolo2#270]
Hi,

Trying to train full yolo on this dataset https://github.com/mr8bit/vedai/blob/master/images/train/100.jpg

(Quite small images of vehicles from aerial/satellite photography)

While loss is trending down nicely, recall settles at 0 after a few epochs and doesnt increase.

I've performed the warm up, and set anchors.

When trying to predict, 0 boxes are found.

Config looks like this:

{
    "model" : {
        "backend":              "Full Yolo",
        "input_size":           416,
        "anchors":              [0.88,1.69, 1.18,0.7, 1.65,1.77,1.77,0.9, 3.75, 3.57],
        "max_box_per_image":    5,
        "labels":               ["car", "truck", "pickup", "tractor", "camping car", "boat","motorcycle", "van"$
    },
    "train": {
        "train_image_folder":   "train_image_folder/",
        "train_annot_folder":   "train_annot_folder/",     
          
        "train_times":          1,
        "pretrained_weights":   "full_yolo_trucks.h5",
        "batch_size":           10,
        "learning_rate":        0.1e-5,
        "nb_epochs":            20,
        "warmup_epochs":        0,
        "object_scale":         5.0,
        "no_object_scale":      2.0,
        "coord_scale":          1.0,
        "class_scale":          1.0,
        "saved_weights_name":   "full_yolo_trucks.h5",
        "debug":                true
    },
    "valid": {
        "valid_image_folder":   "",
        "valid_annot_folder":   "",
        "valid_times":          1
    }
}

Any thoughts? If i want to encourage the model to make predictions, should i set no_object_scale higher?

Thanks!

**Comment by RichardSieg** [comment]
20 epochs is definitely too few. Also you shouldn't name your `pretrained_weights` and the `saved_weights_name` the same. i guess the first one is your warmup trainnig. How many epochs did you use there? For me, usually 1 is enough. You should also consider increasing your batch size, depending on the size of your training set.

**Comment by ZacharyForrest** [comment]
Hello Richard, thanks for your comments!

I named the same as i didnt mind overwriting them (probably bad practice)

For my warm up, im running 3 epochs typically.

So you think if i continue training beyond 20 epochs, recall may start to increase?

Re Batch size - i understand this would be reduce noise in lowering the loss - but is it likely to improve recall?

Appreciate your time

**Comment by ZacharyForrest** [comment]
So i restarted training with just the full yolo backend (no pretrained weights) with a batch size of 16. Loss was trending down and recall trending up til epoch 9, at which point recall started to reduce.

Makes me thing one of the object_scale/no_object scale etc is driving the model to not make predictions

**Comment by ZacharyForrest** [comment]
Hey folks,

So basically i tooled around with the object_scale/class_scale etc and it's now working...just about

In this case i changed class scale to 0.3, which i think means the model is tuned to care less about picking the right class (ie class loss is lower) 

worth saying my dataset is very weighted in favour of one class

**Comment by letilessa** [comment]
Hi @ZacharyForrest, how did you do the warmup stage? What were your final values for object_scale, no_object_scale, coord_scale and class_scale? Can you have a look at my issue #343?

**Comment by ZacharyForrest** [comment]
Hey,

my config is here:
https://github.com/departmentfortransport/dftlab-yolo-vehiclecounting/blob/master/config.json
(note class is 0 cos we were looking for one class)

one thing i vaguely rememer in the notebook is you need to set warm up
batches to 3 epochs worth or so. So for example, if you have batch size 10
for 100 image data set, you need to set warm up batches to 30

Ultaimtely i had more success with the non notebook version!

z

**Comment by letilessa** [comment]
So I only need to train the model for 3 epochs to do the warmup stage?

**Comment by ZacharyForrest** [comment]
yep 3 worked fine for us
